tests/test_consumer.py

```diff
@@ -346,35 +345,28 @@ def test_dead_letter_step() -> None:
 
 def test_metrics_writing_e2e() -> None:
```

Comment:
I would actually be fine removing this test entirely, but it does mean we'd have a bit less coverage of the consumer flow